script-templates/README.md

```diff
@@ -9,6 +13,146 @@ Each of the script templates below can be used as a starting point for developin
+
+The Skytap Metadata Service provides read-only metadata about the Skytap environment hosting an end user's lab. The Course Manager Control Endpoint provides metadata oriented around the end user lab itself, and it also allows limited modifications of the metadata and state of the lab.
+
+The Metadata Service and Control Endpoint can be accessed from within your scripts using HTTP API calls as described below. In addition, the Ruby, Node and Python script templates provide language-specific libraries that wrap the HTTP APIs, making it easier to consume these services. Please consult the README file for the respective template for details.
+
+### Skytap Metadata Service
+
+Access the Metadata Service using the URL `http://skytap-metadata/skytap` from your script. This is a special URL that only works from within Course Manager scripts.
+
+#### Getting the Skytap Metadata
+
+Request:
+```
+GET http://skytap-metadata/skytap
+Accept: application/json
+```
+
+Response:
+```
+200 OK
+
+{"id":"11111111", "name":"Windows Server 2019 Datacenter", "user_data":"{\"control_url\":\"https://customername.skytap-portal.com/lab_access/self_learner/360/.../control/...\"}", ...}
+```
+
+### Lab Control Endpoint
+
+To use the Lab Control Endpoint, your script must first retrieve its URL from the Skytap metadata. Get the Skytap metadata from the Metadata Service as described above and parse the JSON payload. Extract the `user_data` attribute, and parse the JSON string it contains. From there, you can extract the `control_url` attribute, which represents the Lab Control Endpoint.
+
+#### Getting the Lab Control metadata
+
+Request:
+```
+GET https://customername.skytap-portal.com/lab_access/self_learner/360/.../control/...
+Accept: application/json
+
+Response:
+200 OK
+
+{"id":360, "consumed_at":null, "user_identifier":"user@domain.com", "user_access_url":"https://customername.skytap-portal.com/lab_access/self_learner/360/...", ... }
+```
+
+#### Updating Custom Data
+
+Request:
+```
+PUT https://customername.skytap-portal.com/lab_access/self_learner/360/.../control/...
+Content-Type: application/json
+Accept: application/json
+
+{"integration_data": {"AcmeDataProUsername":"user_assigned_from_script","AcmeDataProPassword":"password_assigned_from_script"}}
+```
+
+Response:
+```
+200 OK
+
+{"id":360, "consumed_at":null, "user_identifier":"user@domain.com", "user_access_url":"https://customername.skytap-portal.com/lab_access/self_learner/360/...", ... }
+```
+
+Please note:
+* Custom data fields must be created on the Admin > Settings page (under Labs > Integrations > Custom Data) before they can be updated.
+* Updating custom data overwrites all existing integration data for the lab. If you wish to only update a subset of the integration data fields, retrieve the old integration data, merge your changes in, and then update with the result.
+
+#### Changing Runstate
+You can set the runstate to "running", "suspended", "halted" or "stopped" as follows:
```

Comment:
Link to Skytap help for definition of these runstates? https://help.skytap.com/API_Documentation.html#Run/Stop